### ISSUE TITLE
dub package installation problems under Ubuntu 16.04

### ISSUE THREAD

**Issue by crimaniak** [MikeWey/DMagick#19]
Add dmagick to dependencies to my dub.json.

Problem #1: build-aux/magickVersion.sh is not runnable so we have 'Permission denied'.
Fix: chmod +x ~/.dub/packages/dmagick-0.2.0/build-aux/magickVersion.sh

Problem #2: it tried to compile Windows version and failed:
dmagick 0.2.0: building configuration "other"...
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(23,13): Error: undefined identifier 'WNDCLASS'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(24,13): Error: undefined identifier 'HINSTANCE'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(25,13): Error: undefined identifier 'BITMAPINFO'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(26,13): Error: undefined identifier 'HWND'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(27,13): Error: undefined identifier 'MSG'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(29,22): Error: undefined identifier 'HWND'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(113,41): Error: undefined identifier 'LRESULT'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(113,41): Error: undefined identifier 'HWND'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(113,41): Error: undefined identifier 'UINT'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(113,41): Error: undefined identifier 'WPARAM'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(113,41): Error: undefined identifier 'LPARAM'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(214,12): Error: undefined identifier 'DWORD'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(215,11): Error: undefined identifier 'UINT'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(217,22): Error: undefined identifier 'BOOL'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(217,22): Error: undefined identifier 'HDC'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(217,22): Error: undefined identifier 'HDC'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(217,22): Error: undefined identifier 'DWORD'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(218,25): Error: undefined identifier 'HBITMAP'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(218,25): Error: undefined identifier 'HDC'
../../.dub/packages/dmagick-0.2.0/dmagick/dmagick/internal/Windows.d(219,25): Error: undefined identifier 'HBITMAP'
dmd failed with exit code 1.

**Comment by MikeWey** [comment]
1. The file in git is already executable, maybe dub check it out in a way that that isn't the case?
the output of `git ls-tree HEAD` shows:
```
100755 blob 3a43d05fea38ca8d20a835cf0913a7791095ae33	magickVersion.sh
```

2. I did have some issues with the dub file when trying `dub test`, but is couldn't reproduce your issue with dmd 2.074.0.

**Comment by crimaniak** [comment]
I don't know how dub does it. All I can say it's the fresh install (I never use this library before), dub 1.3.0, DMD64 v2.074.0

**Comment by vitalfadeev** [comment]
Solution. 
In `dmagick/dmagick/internal/Windows.d`
add
```
version (Windows) {
...
}

```

some like this:

```
module dmagick.internal.Windows;

version (Windows) {
import core.sys.windows.windows;

import dmagick.Image;
import dmagick.Exception;
import dmagick.Geometry;
...
extern(Windows) UINT_PTR SetTimer(HWND, UINT_PTR, UINT, TIMERPROC);
}

```

**Comment by MikeWey** [comment]
While that is a good idea, the Windows configuration will only work on Windows.

I misread crimaniak  comment and thought he was trying to build DMagick on windows error 2.

**Comment by MikeWey** [comment]
I've changed the dub.json so that it explicitly makes the auxiliary executable. I think that should get things working again.

**Comment by crimaniak** [comment]
Fix confirmed for version 0.2.1. Thanks! Closing.
